### PR TITLE
Partial EC support

### PR DESCRIPTION
**Pull request by D-o-c**
Hi

I've added partial EC and Ed support based on [this draft](https://tools.ietf.org/html/draft-ietf-openpgp-rfc4880bis-02) of the 4880. It is possible only to read key that are using EC and Ed algorithm but not to create (or sign) one.

Let me know if there are any problems :)
Mattia

### CODE REVIEW

**Comment by D-o-c** [comment]
There is some problem with the test. Soon I'll see it through

File: PKA/PKAs.cpp [hunk 1]
```diff
@@ -6,13 +6,22 @@ namespace PKA {
 bool can_encrypt(const uint8_t alg){
     return ((alg == ID::RSA_ENCRYPT_OR_SIGN) ||
             (alg == ID::RSA_ENCRYPT_ONLY)    ||
-            (alg == ID::ELGAMAL));
+            (alg == ID::ELGAMAL)
+#ifdef GPG_COMPATIBLE
+         || (alg == ID::ECDH)
```
Comment (calccrypto) [inline PKA/PKAs.cpp]: Please maintain the same style as the rest of the code

File: PKA/PKAs.h [hunk 2]
```diff
@@ -67,8 +72,14 @@ namespace OpenPGP {
             std::make_pair(ID::RSA_SIGN_ONLY,       "RSA Sign-Only"),          // deprecated
             std::make_pair(ID::ELGAMAL,             "ELGAMAL (Encrypt-Only)"),
             std::make_pair(ID::DSA,                 "DSA"),
+#ifdef GPG_COMPATIBLE
+            std::make_pair(ID::ECDH,                 "ECDH"),
```
Comment (calccrypto) [inline PKA/PKAs.h]: Please maintain aligntment

File: PKA/PKAs.h [hunk 3]
```diff
@@ -84,12 +95,60 @@ namespace OpenPGP {
             std::make_pair(110,                     "Private/Experimental algorithm"),
         };
 
+
+#ifdef GPG_COMPATIBLE
```
Comment (calccrypto) [inline PKA/PKAs.h]: Please make all `GPGP_COMPATIBLE` if guards be indented with the code

**Comment by calccrypto** [comment]
Can you add some tests to the `tests/` directory? It would be good to have some tests that verify at least the top level functions.

**Comment by calccrypto** [comment]
On a similar note, can you add some tests for your key merging code

**Pull request merged**